kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/ModemTaskScheduler.java

```
@@ -0,0 +1,191 @@
+    private AtomicBoolean isResetScheduled = new AtomicBoolean(false);
+    private int delay = 0;
+
+    public ModemTaskScheduler(String deviceId, Device device, NetworkProperties properties) {
+
+        try {
+            this.nmDbusConnector = Optional.of(NMDbusConnector.getInstance());
+        } catch (DBusExecutionException | DBusException e) {
+            logger.error("Cannot initialize NMDbusConnector due to: ", e);
+        }
+
+        this.device = Objects.requireNonNull(device);
+        this.resetDelayMinutes = properties.get(Integer.class, "net.interface.%s.config.resetTimeout", deviceId);
+        this.autoconnect = properties.get(Boolean.class, "net.interface.%s.config.persist", deviceId);
+        this.holdoff = properties.get(Integer.class, "net.interface.%s.config.holdoff", deviceId);
+        this.maxFail = properties.get(Integer.class, "net.interface.%s.config.maxFail", deviceId);
+        this.deviceId = deviceId;
+
+        this.delay = this.holdoff != 0 && this.maxFail != 0 ? this.holdoff * this.maxFail : 90;
+    }
+
+    public void scheduleConnection() {
+        if (isModemConnectedAndConnectionActivated()) {
+            logger.info("Connection for modem {} successful", this.deviceId);
+            return;
+        }
+        if (isConnectionScheduled.get() || !this.autoconnect) {
+            return;
+        }
+        logger.info("Schedule connection for modem {} with path {}", this.deviceId, this.device.getObjectPath());
+        this.isConnectionScheduled.set(true);
+        this.connectionHandler = this.scheduler.schedule(() -> tryConnection(1), 0, TimeUnit.SECONDS);
+    }
+
+    private void tryConnection(int attemptNumber) {
+        try {
+            logger.debug("Connection attempt {} for modem {} with path {} ...", attemptNumber, this.deviceId,
+                    this.device.getObjectPath());
+            if (this.nmDbusConnector.isPresent()) {
+                this.nmDbusConnector.get().apply(deviceId);
+            }
```

If the `nmDbusConnector` is not present is this going to just fail silently? I would add a log here

```suggestion
            if (this.nmDbusConnector.isPresent()) {
                this.nmDbusConnector.get().apply(deviceId);
            } else {
                logger.warn("Could not attempt reconnection due to...");
            }
```